pandas/tests/groupby/test_raises.py

```diff
@@ -4,23 +4,58 @@
 
 import datetime
 
+import numpy as np
 import pytest
 
-from pandas import DataFrame
+from pandas import (
+    Categorical,
+    DataFrame,
+    Grouper,
+    Series,
+)
 from pandas.tests.groupby import get_groupby_method_args
 
 
+@pytest.fixture(
+    params=[
+        "a",
+        ["a"],
+        ["a", "b"],
+        Grouper(key="a"),
+        lambda x: x % 2,
+        [0, 0, 0, 1, 2, 2, 2, 3, 3],
+        np.array([0, 0, 0, 1, 2, 2, 2, 3, 3]),
+        dict(zip(range(9), [0, 0, 0, 1, 2, 2, 2, 3, 3])),
+        Series([1, 1, 1, 1, 1, 2, 2, 2, 2]),
+        [Series([1, 1, 1, 1, 1, 2, 2, 2, 2]), Series([3, 3, 4, 4, 4, 4, 4, 3, 3])],
+    ]
+)
+def by(request):
+    return request.param
+
+
+@pytest.fixture(params=[True, False])
+def groupby_serie(request):
+    return request.param
+
+
 @pytest.mark.parametrize("how", ["method", "agg", "transform"])
-def test_groupby_raises_string(how, groupby_func, as_index, sort):
+def test_groupby_raises_string(how, by, groupby_serie, groupby_func, as_index, sort):
     df = DataFrame(
         {
-            "a": [1, 1, 1, 2, 2],
-            "b": range(5),
-            "c": list("xyzwt"),
+            "a": [1, 1, 1, 1, 1, 2, 2, 2, 2],
+            "b": [3, 3, 4, 4, 4, 4, 4, 3, 3],
+            "c": range(9),
+            "d": list("xyzwtyuio"),
         }
     )
     args = get_groupby_method_args(groupby_func, df)
-    gb = df.groupby("a", as_index=as_index, sort=sort)
+    gb = df.groupby(by=by, as_index=as_index, sort=sort)
+
+    if groupby_serie:
+        if groupby_func == "corrwith":
+            pytest.skip()
```

Comment:
Why skip here?

Reply:
It seems that the corrwith method is not implemented for SeriesGroupBy. We could test for an AttributeError, but it makes more sense to me to skip it.